src/components/KTooltip/KTooltip.vue

```diff
@@ -25,74 +25,61 @@
   </KPop>
 </template>
 
-<script lang="ts">
-import { defineComponent, computed, ref } from 'vue'
-import KPop from '@/components/KPop/KPop.vue'
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import KPop, { placements } from '@/components/KPop/KPop.vue'
 
-export default defineComponent({
-  name: 'KTooltip',
-  components: { KPop },
-  inheritAttrs: false,
-  props: {
-    /**
+const props = defineProps({
+  /**
      * Text to show in tooltip
      */
-    label: {
-      type: String,
-      required: false,
-      default: '',
-    },
+  label: {
+    type: String,
+    required: false,
+    default: '',
+  },
 
-    /**
-     * Define which side the tooltip displays<br>
-     * 'top' | 'bottom' | 'left' | 'right'
+  /**
+     * Define which side the tooltip displays
      */
-    placement: {
-      type: String,
-      default: 'bottom',
-      validator: (value: string):boolean => {
-        return ['top', 'bottom', 'left', 'right'].includes(value)
-      },
+  placement: {
+    type: String,
+    default: 'bottom',
+    validator: (value: string):boolean => {
+      return Object.keys(placements).includes(value)
```

Comment:
Note for reviewers: besides this line and import of `placements` on line 30, all changes in `<script>` tag are due to switch to `setup` syntax.